src/teachy_tv.c

```diff
@@ -0,0 +1,1571 @@
+        TeachyTvSetWindowRegs();
+        schedule_bg_copy_tilemap_to_vram(0);
+        ChangeBgX(3, 0x0, 0);
+        ChangeBgY(3, 0x0, 0);
+        ChangeBgX(3, 0x1000, 2);
+        ChangeBgY(3, 0x2800, 1);
+        gUnknown_203F450->var_4004 = 0;
+        gUnknown_203F450->var_4005 = 3;
+        gUnknown_203F450->var_4006 = 0;
+    }
+}
+
+void TeachyTvChainTaskBattleOrFadeByOptionChosen(u8 taskId)
+{
+    int op = gTeachyTV_StaticResources.optionChosen;
+    if (op >= 0)
+    {
+        if (op <= 3)
+        {
+            TeachyTvPrepBattle(taskId);
+
+        }
+        else if (op <= 5)
+        {
+            *((void(**)())gUnknown_203F450) = TeachyTvSetupBagItemsByOptionChosen;
+            TeachyTvQuitBeginFade(taskId);
+        }
+    }
+}
+
+void TeachyTvSetupBagItemsByOptionChosen(void)
+{
+    if (gTeachyTV_StaticResources.optionChosen == 4)
+        sub_810B108(10);
+    else
+        sub_810B108(9);
+}
+
+void TeachyTvPostBattleFadeControl(u8 taskId)
+{
+    s16 *data = gTasks[taskId].data;
+    if (!(gPaletteFade.active))
+    {
+        const u8* funcIdx = gUnknown_8479390;
+        int arg = funcIdx[gTeachyTV_StaticResources.optionChosen];
+        data[3] = arg;
+        gTasks[taskId].func = TeachyTvRenderMsgAndSwitchClusterFuncs;
+    }
+}
+
+void TeachyTvGrassAnimationMain(u8 taskId, s16 x, s16 y, u8 subpriority, bool8 mode)
+{
+    u8 subprio;
+    int res;
+    struct Sprite *obj;
+    int objId;
+    const struct SpriteTemplate *const *objTemAddr;
+
+    subprio = subpriority;
+    if (gUnknown_203F450->var_4006 != 1)
+    {
+        res = TeachyTvGrassAnimationCheckIfNeedsToGenerateGrassObj(x - 0x10, y);
+        if (res << 24)
+        {
+            objTemAddr = gUnknown_83A0010;
+            objId = CreateSprite(*(objTemAddr+4), 0, 0, subprio);
+            obj = gSprites + objId;
+            obj->pos2.x = x;
+            obj->pos2.y = y + 8;
+            obj->callback = TeachyTvGrassAnimationObjCallback;
+            obj->data[0] = taskId;
+            if (mode == 1)
+            {
+                SeekSpriteAnim(obj, 4);
+                obj->oam.priority = 2;
+            }
+            else
+            {
+                SetSubspriteTables(obj, gUnknown_84795B8);
+                obj->subspriteTableNum = 0;
+                obj->subspriteMode = 1;
+            }
+        }
+    }
+}
+
+void TeachyTvGrassAnimationObjCallback(struct Sprite *sprite)
+{
+    u32 diff1, diff2;
+    s16 *data = gTasks[sprite->data[0]].data;
+    struct Sprite *objAddr = &gSprites[data[1]];
+    if (gUnknown_203F450->var_4006 == 1)
+        DestroySprite(sprite);
+    else {
+        if (!sprite->animCmdIndex)
+            sprite->subspriteTableNum = 1;
+        else
+            sprite->subspriteTableNum = 0;
+        sprite->pos2.x += (u16)data[4];
+        sprite->pos2.y += (u16)data[5];
+        if (!sprite->animEnded)
+            return;
+        sprite->subpriority = 0;
+        diff1 = (u16)(sprite->pos2.x - objAddr->pos2.x);
```

Comment:
this is all weird and does not look like gamefreak code at all

Reply:
yes, I wasted tons of time on it last time. Again it's like switching from u16 to s16. I fixed that `diff2 += 0xF` issue tho